File: app/src/main/java/grey/smarthouse/database/AppDatabase.java
package grey.smarthouse.database;

import androidx.room.Database;
import androidx.room.RoomDatabase;

import java.util.List;
import java.util.UUID;

import grey.smarthouse.model.Relay;
import grey.smarthouse.model.Repository;

@Database(entities = {Relay.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase implements Repository {
    public abstract RelayDao mRelayDao();


    @Override
    public void insert(Relay relay) {
        mRelayDao().insert(relay);
    }

    @Override
    public void update(Relay relay) {
        mRelayDao().update(relay);
    }

    @Override
    public Relay get(UUID id) {
        return mRelayDao().getById(id);
    }

    @Override
    public List<Relay> getAll() {
        return mRelayDao().getAll();
    }

    @Override
    public void delete(Relay relay) {
        mRelayDao().delete(relay);
    }
}

File: app/src/main/java/grey/smarthouse/database/AppDatabase.kt
package grey.smarthouse.database

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters
import grey.smarthouse.database.converters.IdConverter
import grey.smarthouse.model.Relay
import grey.smarthouse.model.Repository
import java.util.*

@Database(entities = [Relay::class], version = 1)
@TypeConverters(IdConverter::class)
abstract class AppDatabase : RoomDatabase(), Repository {
    abstract fun mRelayDao(): RelayDao


    override fun insert(relay: Relay) {
        mRelayDao().insert(relay)
    }

    override fun update(relay: Relay) {
        mRelayDao().update(relay)
    }

    override fun get(id: UUID): Relay {
        return mRelayDao().getById(id)
    }

    override fun getAll(): List<Relay> {
        return mRelayDao().getAll()
    }

    override fun delete(relay: Relay) {
        mRelayDao().delete(relay)
    }
}
